addons/web/static/src/js/services/crash_manager.js

```diff
@@ -4,27 +4,110 @@ odoo.define('web.CrashManager', function (require) {
 var ajax = require('web.ajax');
 var core = require('web.core');
 var Dialog = require('web.Dialog');
+var Widget = require('web.Widget');
 
-var QWeb = core.qweb;
 var _t = core._t;
 var _lt = core._lt;
 
 var map_title ={
-    user_error: _lt('Warning'),
-    warning: _lt('Warning'),
-    access_error: _lt('Access Error'),
-    missing_error: _lt('Missing Record'),
-    validation_error: _lt('Validation Error'),
-    except_orm: _lt('Global Business Error'),
-    access_denied: _lt('Access Denied'),
+    user_error: _lt("User Error"),
+    warning: _lt("Warning"),
+    access_error: _lt("Access Error"),
+    missing_error: _lt("Missing Record"),
+    validation_error: _lt("Validation Error"),
+    except_orm: _lt("Global Business Error"),
+    access_denied: _lt("Access Denied"),
 };
 
+/**
+ * An extension of Dialog Widget to render the warnings and errors on the website.
+ * Extend it with your template of choice like ErrorDialog/WarningDialog
+ */
+var CrashManagerDialog = Dialog.extend({
+    xmlDependencies: (Dialog.prototype.xmlDependencies).concat(
+        ['/web/static/src/xml/crash_manager.xml']
+    ),
+    /**
+     * @param {Object} error
+     * @param {string} error.message    the message in Warning/Error Dialog
+     * @param {string} error.traceback  the traceback in ErrorDialog
+     */
+    init: function (parent, options, error) {
+        this._super.apply(this, [parent, options]);
+        this.message = error.message;
+        this.traceback = error.traceback;
+    },
+});
+
+var ErrorDialog = CrashManagerDialog.extend({
+    template: 'CrashManager.error',
+});
+
+var WarningDialog = CrashManagerDialog.extend({
+    template: 'CrashManager.warning',
+    /**
+     * set size to medium
+     * @override
+     */
+    init: function (parent, options) {
+        options.size = 'medium',
```

Comment:
Do not modify a parameter object in place. Also what if the user want to instantiate a warning dialog with large size (dummy case but the parameters is there to allow that).
Something like this would be more correct
```javascript
this._super(parent, _.extend({
     size: 'medium',
}, options));
```